storage/txcache/eviction.go

```diff
@@ -7,78 +7,91 @@ import (
 // EvictionConfig is a cache eviction model
 type EvictionConfig struct {
 	Enabled                         bool
+	NumBytesThreshold               uint32
 	CountThreshold                  uint32
 	ThresholdEvictSenders           uint32
-	NumOldestSendersToEvict         uint32
+	NumSendersToEvictInOneStep      uint32
 	ALotOfTransactionsForASender    uint32
 	NumTxsToEvictForASenderWithALot uint32
 }
 
 // doEviction does cache eviction
 // We do not allow more evictions to start concurrently
-func (cache *TxCache) doEviction() evictionJournal {
-	if !cache.areThereTooManyTxs() {
-		return evictionJournal{}
+func (cache *TxCache) doEviction() {
+	tooManyBytes := cache.areThereTooManyBytes()
+	tooManyTxs := cache.areThereTooManyTxs()
+	tooManySenders := cache.areThereTooManySenders()
+	journal := evictionJournal{}
+
+	if !tooManyBytes && !tooManyTxs && !tooManySenders {
```

Comment:
take out into a variable named txPoolIsOk = ... it is easier to understand.

Reply:
I've rewritten the condition and used the new variable `isCapacityExceeded`.